### DISCUSSION TITLE
Style standardization

### DISCUSSION THREAD

**Discussion by omor1** [umurmur/umurmur#198] [General]
#63 opened up a style standardization discussion, starting with NULL pointer checking.

Please post any style discrepancies found so that a single style standard for umurmur may be decided upon (and preferably collected in a document to be put in the wiki).

**Comment by omor1** [comment]
NULL pointer checking

``` C
char* pointer = NULL
if (!pointer)
if (pointer == NULL)
if (NULL == pointer)
```

**Comment by fmorgner** [comment]
Location of local variable declarations

``` c
void foo(...)
    {
    int bla;
    char blob;
    \\ a lot of code here (not related to bla and blob)
    bla = bar(...);
    blob = baz(bla);
    \\ maybe some more code
    }
```

vs.

``` c
void foo(...)
    {
    \\ a lot of code here (not related to bla and blob)
    int bla = bar(...);
    char blob = baz(bla);
    \\ maybe some more code
    }
```

**Comment by fmorgner** [comment]
Single line if (the same goes for else)

``` c
if(condition)
    do_stuff();
```

vs.

``` c
if(condition) {
    do_stuff();
}
```

which is interestingly enough interpreted differently by the GitHub Markdown interpreter :laughing:

**Comment by Diaoul** [comment]
For style standardization I use [astyle](http://astyle.sourceforge.net/) for my personal projects. There are a lot of options.
